odpc.client/src/features/publicatie/composables/use-paged-search.ts

```diff
@@ -0,0 +1,121 @@
+import { ref, watch, computed, onMounted, type Ref } from "vue";
+import { useUrlSearchParams, type UrlParams } from "@vueuse/core";
+import { useFetchApi } from "@/api/use-fetch-api";
+
+const API_URL = `/api/v1`;
+const PAGE_SIZE = 5;
+
+export type PagedResult<T> = {
+  count: number;
+  next?: string;
+  previous?: string;
+  results: T[];
+};
+
+export const usePagedSearch = <T, U extends string>(endpoint: string, params: UrlParams) => {
+  type QueryParams = Record<U, string> & { page: string; search: string };
+  type QueryParam = keyof QueryParams;
+
+  const requiredParams = { page: "", search: "" };
+
+  params = { ...requiredParams, ...params };
+
+  const paramKeys = Object.keys(params) as QueryParam[];
+
+  const urlSearchParams = useUrlSearchParams("history", {
+    initialValue: params,
+    removeFalsyValues: true
+  });
+
+  const pagedResult = ref(null) as Ref<PagedResult<T> | null>;
+
+  const queryParams = ref(params) as Ref<QueryParams>;
+  const searchString = ref("");
+
+  const initQueryParams = () => {
+    queryParams.value = paramKeys.reduce(
+      (result, key) => ({ ...result, [key]: urlSearchParams[key] || "" }),
+      {} as QueryParams
+    );
+  };
+
+  const onNext = () => {
+    queryParams.value.page = `${+queryParams.value.page + 1}`;
+  };
+
+  const onPrev = () => {
+    queryParams.value.page = `${+queryParams.value.page - 1}`;
+  };
+
+  const onSearch = () => (queryParams.value = { ...queryParams.value, search: searchString.value });
```

Comment:
de searchString toevoegen aan de queryparams , om daar vervolgens een watch op te zettten, die weer searchString vult. waarom is dat nodig?
vervolgens wordt er weer een andere computed gevuld obv de queryParams en dan zit daar weer een watch op die de daadwerkelijke search gaat doen. 
Er is an sich niets mis mee, maar is het niet wat over complex? 
als het niet simpeler kan dan svp wat meer toelichting in de code, denk aan: wat is de functie van de watches, het verschil tussen searchparams en queryparams.

Reply:
Heb comments toevoegd over werking van deze reactive props.